packages/goto/scripts/postinstall.js

```diff
@@ -1,13 +1,16 @@
 'use strict'
 
-const { parseFilters } = require('@cliqz/adblocker')
+const { PuppeteerBlocker, getLinesWithFilters } = require('@cliqz/adblocker')
 const { promisify } = require('util')
 const { EOL } = require('os')
 const got = require('got')
 const fs = require('fs')
 
 const writeFile = promisify(fs.writeFile)
 
+// uBlock Origin –  https://github.com/uBlockOrigin/uAssets/tree/master/filters
+const RESOURCES = 'https://raw.githubusercontent.com/uBlockOrigin/uAssets/master/filters/resources.txt'
```

Comment:
Can you explain how different are `resources` compared with `filters`? 🤔 

Also, I tried to add in the past sensible good defaults, not sure if I'm missing a good addition we can do here

Reply:
`resources.txt` is needed to make use of some kind of filters. There are mostly two cases here:

1. Some filters are called `redirect`. Instead of blocking a request if it matches the filter, it will instead redirect it to a local resource. For example a rule like `pixel.gif$redirect=1x1-transparent.gif` might be matching any request with URL ending with `pixel.gif` (most likely a tracking, transparent pixel) and return a mock of a fake transparent pixel (in this case `1x1-transparent.gif`). This particular resource needs to be defined in `resources.txt`! Here is how the resource looks like: https://github.com/uBlockOrigin/uAssets/blob/master/filters/resources.txt#L13. You can check how it's implemented for Puppeteer here: https://github.com/cliqz-oss/adblocker/blob/master/src/puppeteer/blocker.ts#L102

Basically it prevents any network from happening but instead answers with a pre-defined constant `string` as `body` for this particular request. Another common use case would be to prevent Google Analytics to load into any page, but blocking the network request could potentially break the page, or allow the website to detect that an adblocker is being used; instead, a redirect filter can be used to send a "mocked" version of the Google Analytics script (without any real logic), exposing the same API. You can see this particular example here: https://github.com/uBlockOrigin/uAssets/blob/master/filters/resources.txt#L252

2. There are filters called `cosmetics` injections, which can inject small pieces of self-contained JavaScript (a.k.a.: "scriptlets") into pages (this is done via the `Frame.addScriptTag(...)` API of puppeteer and you can see it there: https://github.com/cliqz-oss/adblocker/blob/master/src/puppeteer/blocker.ts#L76). One such filter could be `example.com##+js(,mangareader.net,toonget.net,toonova.net##+js(set-constant.js, String.fromCharCode, trueFunc)`; this would run the `set-constant.js` sciptlet into pages of domains `example.com`; this mocks `String.fromCharCode` with a function returning a constant. These are used to prevent sites from detecting adblocker for example.